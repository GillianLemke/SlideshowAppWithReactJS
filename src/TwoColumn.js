import React from 'react';
//import { createStore } from 'redux';
//import ReactDOM from 'react-dom';
import './index.css';
import $ from 'jquery';

//let store = createStore()
export default class TwoColumn extends React.Component {
  getPageInfo() {
    return $.getJSON('google.com/slide')
      .then((data) => {
        // how can we get data for individual variables?
        this.setState({ title:data.results })
      });
  }
  makeColumn(contentList) {
    return(
      <ul>
        {contentList.map((item, index) => <li key={index}>{item}</li>)}
      </ul>
    );
  }
  render() {
    return (
      <div className="two-column-slide">
        <h1>{this.props.twoColumnData.title}</h1>
        <table>
          <tbody>
            <tr>
              <td>
                {this.makeColumn(this.props.twoColumnData.content)}
              </td>
              <td>
                {this.makeColumn(this.props.twoColumnData.content2)}
              </td>
            </tr>
          </tbody>
        </table>

      </div>
    );
  }

}
