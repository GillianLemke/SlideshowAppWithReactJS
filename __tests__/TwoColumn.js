import React from 'react'
import TwoColumn from '../src/TwoColumn.js'
import {shallow} from 'enzyme' // could also mount
import toJson from 'enzyme-to-json'
import * as ShallowTestUtils from 'react-shallow-testutils';

describe('TwoColumn', () => {
  it('renders', () => {
    const wrapper = shallow(<TwoColumn title='' type='' contents={[]} contents2={[]} />)
    expect(toJson(wrapper)).toMatchSnapshot()
  });
  // it('has two columns', () => {
  //   // how to properly test for two columns
  //   const list1 = ['hello', 'bonjour']
  //   const list2 = ['goodbye', 'ciao']
  //   const view = ShallowTestUtils.renderIntoDocument(<TwoColumn />)
  //   const resultFromCol1 = view.makeColumn1(list1)
  //   const resultFromCol2 = view.makeColumn2(list2)
  //   const wrapper = shallow(<TwoColumn title='' type='' contents={[]} contents2={[]} />)
  //   expect(wrapper.find('li').length) >= 1
  // });
});
