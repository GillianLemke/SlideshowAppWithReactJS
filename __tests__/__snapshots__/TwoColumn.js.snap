exports[`TwoColumn renders 1`] = `
<div
  className="two-column-slide">
  <h1>
    column title
  </h1>
  <table>
    <tbody>
      <tr>
        <td>
          <ul>
            <li>
              col 1
            </li>
          </ul>
        </td>
        <td>
          <ul>
            <li>
              col 2
            </li>
          </ul>
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
